src/fides/api/ops/service/connectors/dynamodb_connector.py

```diff
@@ -0,0 +1,166 @@
+from fides.connectors.models import (
+    AWSConfig,
+    ConnectorAuthFailureException,
+    ConnectorFailureException,
+)
+
+
+class DynamoDBConnector(BaseConnector[Any]):  # type: ignore
+    """AWS DynamoDB Connector"""
+
+    def build_uri(self) -> None:
+        """Not used for this type"""
+
+    def close(self) -> None:
+        """Not used for this type"""
+
+    def create_client(self) -> Any:  # type: ignore
+        """Returns a client for a DynamoDB instance"""
+        config = DynamoDBSchema(**self.configuration.secrets or {})
+        try:
+            aws_config = AWSConfig(
+                region_name=config.region_name,
+                aws_access_key_id=config.aws_access_key_id,
+                aws_secret_access_key=config.aws_secret_access_key,
+            )
+            return aws_connector.get_aws_client(
+                service="dynamodb", aws_config=aws_config
+            )
+        except ValueError:
+            raise ConnectionException("Value Error connecting to AWS DynamoDB.")
+
+    def query_config(self, node: TraversalNode) -> QueryConfig[Any]:
+        """Query wrapper corresponding to the input traversal_node."""
+        client = self.client()
+        try:
+            describe_table = client.describe_table(TableName=node.address.collection)
+            attribute_definitions = describe_table["Table"]["AttributeDefinitions"]
+        except ClientError as error:
+            raise ConnectorFailureException(error.response["Error"]["Message"])
+
+        return DynamoDBQueryConfig(node, attribute_definitions)
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Connects to AWS DynamoDB and lists tables to validate credentials.
+        """
+        logger.info("Starting test connection to {}", self.configuration.key)
+        client = self.client()
+        try:
+            client.list_tables()
+        except ClientError as error:
+            if error.response["Error"]["Code"] in [
+                "InvalidClientTokenId",
+                "SignatureDoesNotMatch",
+            ]:
+                raise ConnectorAuthFailureException(error.response["Error"]["Message"])
+            raise ConnectorFailureException(error.response["Error"]["Message"])
+
+        return ConnectionTestStatus.succeeded
+
+    def retrieve_data(
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        input_data: Dict[str, List[Any]],
+    ) -> List[Row]:
+        """
+        Retrieve DynamoDB data.
+        In the case of complex objects, returns multiple rows
+        as the product of options to query against.
+        """
+        deserializer = TypeDeserializer()
+        collection_name = node.address.collection
+        client = self.client()
+        try:
+            results = []
+            query_config = self.query_config(node)
+            # TODO check this out with a primary key and a sort key
```

Comment:
What's the plan for this action?

Reply:
Tested locally but I don't think moved it into the datasets we are actually testing repeatedly and I didn't remove the comment. Will dig in here